spring-integration-smb/src/main/java/org/springframework/integration/smb/outbound/SmbMessageHandler.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.smb.outbound;
+
+import org.springframework.integration.file.remote.handler.FileTransferringMessageHandler;
+import org.springframework.integration.file.remote.session.SessionFactory;
+import org.springframework.integration.file.support.FileExistsMode;
+import org.springframework.integration.smb.session.SmbRemoteFileTemplate;
+import org.springframework.integration.smb.session.SmbSessionFactory;
+
+import jcifs.smb.SmbFile;
+
+/**
+* The SMB specific {@link FileTransferringMessageHandler} extension. Based on the
+* {@link SmbRemoteFileTemplate}.
+*
+* @author Gregory Bragg
+*
+* @since 1.3
+*
+* @see SmbRemoteFileTemplate
+*/
+public class SmbMessageHandler extends FileTransferringMessageHandler<SmbFile> {
+
+	public SmbMessageHandler(SessionFactory<SmbFile> sessionFactory) {
+		this(new SmbRemoteFileTemplate(sessionFactory));
+		SmbSessionFactory smbSessionFactory = (SmbSessionFactory) sessionFactory;
```

Comment:
I think we need to have a type check here.
There is a `CachingSessionFactory` which may wrap an original one. So, we will fail here with class cast exception.

Reply:
This is kinda why I wanted to clean up this code to remove the unneeded flags and go to a major version... this is getting messy now. When you see the latest edits you will understand what I mean...

Reply:
Yeah...
If you feel so uncomfortable with such a mess, we definitely can look into a new major version.
But that has to go to Spring Integration project already as a module: https://github.com/spring-projects/spring-integration-extensions/issues/254

I just don't want to support two major versions at the same time...

WDYT?

Reply:
Yeah understood,,, I'm working on trying to get this as clean as possible.